src/main/java/edu/indiana/dlib/amppd/service/impl/JobServiceImpl.java

```diff
@@ -295,6 +301,8 @@ public WorkflowOutputs createJob(String workflowId, Long primaryfileId, Map<Stri
     		populateHmgmContextParameters(workflowDetails, primaryfile, winputs.getParameters());
     		msg_param = ", parameters (system updated): " + winputs.getParameters();
     		woutputs = workflowsClient.runWorkflow(winputs);
+    		
+    		dashboardService.addDashboardResult(workflowDetails.getId(), workflowDetails.getName(), primaryfile.getId(), woutputs.getHistoryId());
     	}
```

Comment:
workflowId and primaryfileId is avaiable as the method params. for workflowId, it's safer just use it instead of (workflowDetails.getId(), as there might be a small chance the returned ID is not a stored workflow ID. 

Also, most of the info needed in DashboardResult is directly available in the woutputs returned above, I wonder if you can take advantage of that to reduce querying. However, the dataset visibility is not in the WorkflowOutputs. You could use the WorkflowOutputs.outputIds to query the datasets. If this works, then you probably dont need to query job details.

Reply:
Is ToolId available somewhere other than the job? If not, we need the InvocationDetails to get the InvocationStepDetails so we can get the jobs.